lib/pbench/agent/modules/metaclass.py

```diff
@@ -40,27 +44,32 @@ def getFullData(self):
                 return None
             with self.json.open("r") as json_file:
                 metadata = json.load(json_file)
-                self.data = metadata
         elif self.mode == "redis":
             try:
                 meta_raw = self.redis_server.get("tool-metadata")
                 if meta_raw is None:
-                    self.logger.error("Metadata was never loaded into redis")
+                    if self.initialized:
+                        self.logger.error("Metadata has not been loaded into redis yet")
```

Comment:
This construct bothers me a bit. If we start up an agent tool meister expecting to pull Redis data, and the initial `__getInitialData` in the constructor fails, it'll fail silently, returning `None`; and a subsequent call will simply log that Metadata hasn't been loaded into redis... which isn't strictly true, nor useful.

I assume that the intent was to catch the case where we construct with JSON, and the object gets switched to `redis` mode without calling `loadIntoRedis` ... that seems like a rare edge case, and could be avoided by restricting the mode change to a method that changed mode, loaded the JSON into redis, and failed visibly if anything went wrong.